### PR TITLE
walkdir and read-only modifications

### PR DESCRIPTION
**Pull request by behinger**

### CODE REVIEW

File: src/highlevel.jl [hunk 1]
```diff
@@ -230,3 +239,48 @@ url(f) = url(f, only(view_only_links(project(f))))
 
 Base.read(f::Union{File,FileVersion}) = take!(Downloads.download(string(url(f)), IOBuffer()))
 Base.read(f::Union{File,FileVersion}, ::Type{String}) = String(read(f))
+
+
+
+import Base.walkdir
```
Comment (aplavin) [inline src/highlevel.jl:245]: that's not needed

File: src/highlevel.jl [hunk 2]
```diff
@@ -230,3 +239,48 @@ url(f) = url(f, only(view_only_links(project(f))))
...
+Base.walkdir(proj::Project; kwargs...) = walkdir(directory(proj, ""); kwargs...)
```
Comment (aplavin) [inline src/highlevel.jl:246]: more generic solution that is probably better:
- remove this method
- make `dir::Union{Directory,Project}` in the method below
- add `readdir(::Project)`
It'll work then I think, without a separate walkdir() method.

**Review by aplavin — CHANGES_REQUESTED** [review]
Should be fine with some minor comments!

But please revert all the formatting (whitespace) changes – they are unrelated to this PR.

File: src/highlevel.jl (hunk 2 above)
Comment (behinger) [reply src/highlevel.jl:246]: I resolved this in the other PR. Note that the channel is now a `Channel::Union{Project,Directory})` due to your suggested change

**Comment by behinger** [comment]
I close this PR in favor of #4

**Pull request closed without merging**